### PR TITLE
add GlobalProxySelection

### LINKED ISSUE #12128: Port System.Net.GlobalProxySelection
```csharp
    public class GlobalProxySelection {
        public GlobalProxySelection();
        public static IWebProxy Select { get; set; }
        public static IWebProxy GetEmptyWebProxy();
    }
```

### PR DESCRIPTION
**Pull request by geoffkizer**
Fixes #12128

### CODE REVIEW

**Comment by geoffkizer** [comment]
@stephentoub @ericeil @CIPop @davidsh

**Comment by geoffkizer** [comment]
Builds seem to be hitting #13025

File: src/System.Net.Requests/src/System/Net/GlobalProxySelection.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    [Obsolete("This class has been deprecated. Please use WebRequest.DefaultWebProxy instead to access and set the global default proxy. Use 'null' instead of GetEmptyWebProxy. http://go.microsoft.com/fwlink/?linkid=14202")]
+    public class GlobalProxySelection
+    {
+        // This defers to WebRequest.DefaultWebProxy, but returns EmptyWebProxy instead of null.
+        public static IWebProxy Select
+        {
+            get
+            {
+                IWebProxy proxy = WebRequest.DefaultWebProxy;
+                if (proxy == null)
```
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/GlobalProxySelection.cs]: Will this ever happen in our .NET Core implementation?

Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/GlobalProxySelection.cs]: Oh, I see, it can be explicitly set to null.

**Comment by geoffkizer** [comment]
@dotnet-bot 
Test Innerloop Windows_NT Debug Build and Test
Test Innerloop Windows_NT Release Build and Test

**Pull request merged**